src/components/VForm/VForm.js

```diff
@@ -35,41 +30,80 @@ export default {
 
   methods: {
     getInputs () {
-      return this.$children.filter(child => {
-        return typeof child.errorBucket !== 'undefined'
+      const results = []
+
+      const search = (children, depth = 0) => {
+        for (const child of children) {
+          if (child.errorBucket !== undefined) {
+            results.push(child)
+          } else {
+            search(child.$children, depth + 1)
+          }
+        }
+        if (depth === 0) return results
+      }
+
+      return search(this.$children)
+    },
+    watchInputs (inputs) {
+      inputs === undefined && (inputs = this.getInputs())
+
+      for (const child of inputs) {
+        if (this.inputs.includes(child)) {
+          continue // We already know about this input
+        }
+
+        this.inputs.push(child)
+
+        this.watchChild(child)
+      }
+    },
+    watchChild (child) {
+      const watcher = (child) => {
+        child.$watch('valid', (val) => {
+          this.$set(this.errorBag, child._uid, !val)
+        }, { immediate: true })
+      }
+
+      if (!this.lazyValidation) return watcher(child)
+
+      // Only start watching inputs if we need to
+      child.$watch('shouldValidate', (val) => {
+        if (!val) return
+
+        // Only watch if we're not already doing it
+        if (this.errorBag.hasOwnProperty(child._uid)) return
+
+        watcher(child)
       })
     },
     validate () {
-      const errors = this.getInputs().reduce((errors, child) => {
-        const error = !child.validate(true)
-        return errors || error
-      }, false)
-
-      return !errors
+      return !this.inputs.filter(input => input.validate(true)).length
```

Comment:
This could/should be ```!this.inputs.every(input => input.validate(true))```

Reply:
I had that in 9a5063bf, but it stops at the first invalid field. I can change it to `...length === 0` to make it more clear what is happening.

Reply:
Could also possibly use `this.inputs.some(input => !input.validate(true))`

Reply:
`.some` has the same problem as `.every`. 

> The every method executes the provided callback function once for each element present in the array until it finds one where callback returns a falsy value. If such an element is found, the every method immediately returns false.

> some() executes the callback function once for each element present in the array until it finds one where callback returns a truthy value (a value that becomes true when converted to a Boolean). If such an element is found, some() immediately returns true. 

`input.validate()` must be called on every element so that validation errors are forced to display.